   Johannes Stezenbach [4]js at sig21.net
   Wed Jan 19 04:43:34 EST 2011

Powersaving is broken on rt2800usb (the hardware cannot do timed wakeup
to receive beacons) and also on rt2800pci currently (auto-wakeup irq
does no seem to work, for reasons so far unknown).

Signed-off-by: Johannes Stezenbach <[11]js at sig21.net>

Index: linux-2.6-submit/drivers/net/wireless/rt2x00/rt2800lib.c
===================================================================
--- linux-2.6-submit.orig/drivers/net/wireless/rt2x00/rt2800lib.c	2011-03-11 10:54:51.000000000 +0100
+++ linux-2.6-submit/drivers/net/wireless/rt2x00/rt2800lib.c	2011-03-30 20:49:59.000000000 +0200
@@ -3240,10 +3240,9 @@ int rt2800_probe_hw_mode(struct rt2x00_d
 	u16 eeprom;
 
 	/*
-	 * Disable powersaving as default on PCI devices.
+	 * Disable powersaving as default.
 	 */
-	if (rt2x00_is_pci(rt2x00dev) || rt2x00_is_soc(rt2x00dev))
-		rt2x00dev->hw->wiphy->flags &= ~WIPHY_FLAG_PS_ON_BY_DEFAULT;
+	rt2x00dev->hw->wiphy->flags &= ~WIPHY_FLAG_PS_ON_BY_DEFAULT;
 
 	/*
 	 * Initialize all hw fields.
